app/api/chat/route.ts

```diff
@@ -16,24 +69,103 @@ export async function POST(request: Request) {
   const title =
     message.length > 40 ? message.slice(0, 40).concat("...") : message;
 
-  const saveChatResult = await saveChat({
+  const [chat] = await saveChat({
     userId: session.user.id,
     title,
+    marketAddress: marketId,
   });
-  const chat = saveChatResult[0];
 
   await saveMessage({
     chatId: chat.id,
     message: { response: message },
     role: "user",
   });
 
+  try {
+    if (marketId) createMarketChat(message, chat, marketId);
+    else verifyAndCreateChat(message, chat);
+  } catch (error) {
+    let message;
+    if (error instanceof Error) message = error.message;
+    else message = String(error);
+
+    return new Response(message, { status: 400 });
+  }
+
   return NextResponse.json({ chatId: chat.id });
 }
 
+async function createMarketChat(
+  message: string,
+  chat: Chat,
+  marketAddress: string
+) {
+  const preditcion = await getPredictionByMarketAddress({ marketAddress });
+
+  if (!preditcion) return createChat(message, chat);
```

Comment:
Minor, non blocking:
Small typo in the variable name (there are other ocurrences of this typo that I didn't pick to avoid doing a lengthy suggestion)

```suggestion
  const prediction = await getPredictionByMarketAddress({ marketAddress });

  if (!prediction) return createChat(message, chat);
```